### DISCUSSION TITLE
<x-wireui.select> and dynamic data

### DISCUSSION THREAD

**Discussion by rabol** [wireui/wireui#551] [Q&A]
I'm trying to use the <x-wireui.select> in a form component: I add controls when a button is pushed

NOTE: I have renamed the alias, otherwise WireUI don't play well with Laravel's way of using components

```
<x-wireui.select
    wire:model="recipeIngredients.{{$index}}.ingredient_id"
    placeholder="Select ingredient"
    label="Ingredient"
    class="mt-1 block w-ful "
    :async-data="route('api.ingredient-search')"
    option-label="name"
    option-value="id"
    x-on:selected="$wire.recipeIngredientsChanged()"
    name="recipeIngredients.{{$index}}.ingredient_id"
 />
```

the x:on:selected calls the correct method, but without any arguments, so one does not know the selected value

a standard select works
```
<select wire:model="recipeIngredients.{{$index}}.ingredient_id" name="recipeIngredients.{{$index}}.ingredient_id">
    @foreach($allIngredients as $ing)
          <option value="{{$ing->id}}">{{$ing->name}}</option>
    @endforeach
</select>
```
the selected value is not added to the array like other 'normal' values will be using the wire:model, how can I make that work ?

**Comment by ph7jack** [comment]
You need to have the array keys to work with the x-select
The alpinejs entangle doesn't create the key like the normal wire:model

**Reply by ph7jack** [reply]
See this https://github.com/wireui/wireui/issues/513

**Reply by rabol** [reply]
I'm really sorry, but I can get it to work, even after looking at the reference issues

I have this:
```
<x-wireui.select
    wire:model.defer="recipeIngredient.{{$index}}"
    wire:key="recipe::ingredient::{{$index}}.ingredient_id"
    name="recipeIngredient.{{$index}}.ingredient_id"
    id="recipeIngredient.{{$index}}.ingredient_id"
    placeholder="Select ingredient"
    label="Ingredient"
    class="mt-1 block w-ful "
    :async-data="route('api.ingredient-search')"
    option-label="name"
    option-value="id"
    x-on:selected="$wire.recipeIngredientsChanged()"
    x-on:clear="$wire.recipeIngredientsChanged()"
/>
```

$recipeIngredient is a multi-level array
```
array:1 [▼
  0 => array:10 [▼
    "recipe_ingredient_id" => 1
    "raw_ingredient" => ""
    "is_saved" => false
    "recipe_id" => null
    "ingredient_id" => 0
    "ingredient_unit_id" => 0
    "ingredient_name" => ""
    "unit_name" => ""
    "action_name" => ""
    "action_id" => 0
  ]
]
```
the ingredient_id of the array does not get updated

**Reply by rabol** [reply]
I think I have found the solution.

Still struggling to get the 'selected' value in the x-on:selected

**Reply by ph7jack** [reply]
@rabol You have incongruent values
Your `wire:model` is missing the `ingredient_id` column
When using livewire, the name is just a property to render the error name, the `wire:model` will update the values

TIP: if you pass a `wire:model`, the id and name will be the same as the `wire:model`

![image](https://user-images.githubusercontent.com/38913462/205198354-00acf585-1113-45be-9112-119de60e84de.png)
